yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -59,6 +59,10 @@ private[spark] class ApplicationMaster(
   private val maxNumExecutorFailures = sparkConf.getInt("spark.yarn.max.executor.failures",
     sparkConf.getInt("spark.yarn.max.worker.failures", math.max(args.numExecutors * 2, 3)))
 
+  // Disable the maximum executor failure check
+  private val disableMaxExecutorFailureCheck = 
+              sparkConf.getBoolean("spark.yarn.max.executor.failures.disable", false)
```

Comment:
There's a cost and weight to making a flag for everything, and I think this doesn't add value. Just set max to a high value to "disable" it.